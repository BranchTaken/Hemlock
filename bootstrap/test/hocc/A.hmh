hocc
    left p1
    left p2 < p1

    token PLUS
    token STAR
    token LPAREN
    token RPAREN prec p1
    token ID
    token EOI

    start S ::= E EOI

    nonterm E ::=
      | E PLUS T prec p2
      | T

    nonterm T ::=
      | T STAR F
      | F

    nonterm F prec p1 ::=
      | LPAREN E RPAREN
      | ID
