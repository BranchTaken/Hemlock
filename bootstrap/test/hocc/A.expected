hocc: Parsing "./A.hmh"
hocc: Generating LR(1) specification
hocc: 2 precedences, 8 tokens, 5 non-terminals (1 start), 8 productions
hocc: LR(1) item set compatibility: lr1
hocc: Generating LR(1) item set closures (+^.=add/split/merge)++++++^^+^^++++^^+^++^^^
hocc: Generating 25 LR(1) states
hocc: 0 unresolvable conflicts in 0 states
hocc: 0 unreachable states
hocc: 10 remergeable states
hocc: Reindexing 15 LR(1) states
hocc: Searching for unused precedences/tokens/non-terminals/productions
hocc: Generating text report
hocc: Generating hocc report
hocc: Writing "./hocc/A.txt"
hocc: Writing "./hocc/A.hmh"
